base,testutils,kvserver: disable lease queue in replication manual

kvoli opened the pull request:
A replica lease queue was introduced in https://github.com/cockroachdb/cockroach/pull/119155, which processes lease
transfers for replicas. Previously, the replicate queue handled lease
transfers. The linked change omitted updating the `ReplicationManual`
test cluster mode to disable the lease queue, resulting in unexpected
lease transfers in some tests.

Disable the lease queue under `ReplicationManual` replication mode. Misc
tests are also updated to disable/enable the lease queue appropriately.

Epic: None
Touches: https://github.com/cockroachdb/cockroach/issues/120605
Release note: None

kvoli committed:
base,testutils,kvserver: disable lease queue in replication manual

A replica lease queue was introduced in #119155, which processes lease
transfers for replicas. Previously, the replicate queue handled lease
transfers. The linked change omitted updating the `ReplicationManual`
test cluster mode to disable the lease queue, resulting in unexpected
lease transfers in some tests.

Disable the lease queue under `ReplicationManual` replication mode. Misc
tests are also updated to disable/enable the lease queue appropriately.

Epic: None
Touches: #120605
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/121052)
<!-- Reviewable:end -->

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121052#-:-NtvcXmr80cJ1BBtrU9r:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli)

___
*[`pkg/kv/kvserver/queue_helpers_testutil.go` line 115 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121052#-Ntvc1ig-nvW-V5sH_iv:-Ntvc1ig-nvW-V5sH_iw:b-1limdh) ([raw file](https://github.com/cockroachdb/cockroach/blob/9d9022e2de9659cf2c5988f534bacc51fba2535f/pkg/kv/kvserver/queue_helpers_testutil.go#L115)):*
> ```Go
> }
> 
> // SetLeaseQueueActive controls activating the lease queue. Only intended for
> ```

nit: consider prefixing these (and others) with "Testing" (maybe in a different commit though)


<!-- Sent from Reviewable.io -->

kvoli committed:
cli,kvserver,testutils: prefix queue toggle with testing

The `(s|S)etXQueueActive` test utility functions should only be used in
testing. Prefix them with `Testing` to make this clear.

Epic: none
Release note: None

kvoli reviewed (commented):
TYFTR

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121052#-:-NtvmRHo0b-Kppy4n7H7:b-80sls6)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani and @kvoli)

___
*[`pkg/kv/kvserver/queue_helpers_testutil.go` line 115 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121052#-Ntvc1ig-nvW-V5sH_iv:-NtvmNCNEIua6l9GFAbt:b-8qr0wo) ([raw file](https://github.com/cockroachdb/cockroach/blob/9d9022e2de9659cf2c5988f534bacc51fba2535f/pkg/kv/kvserver/queue_helpers_testutil.go#L115)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: consider prefixing these (and others) with "Testing" (maybe in a different commit though)
</blockquote></details>

Good idea, pushed another commit with this change.


<!-- Sent from Reviewable.io -->

kvoli commented:
bors r=arulajmani
